runtime/permissions.rs

```diff
@@ -63,515 +65,506 @@ impl Default for PermissionState {
+          PermissionState::Granted
         } else {
           self
-            .read
             .denied_list
-            .retain(|path| !resolved_path.starts_with(path));
-          self.read.denied_list.insert(resolved_path);
-          self.read.global_state = PermissionState::Denied;
-          return PermissionState::Denied;
+            .retain(|path| !resolved_path.starts_with(&path.0));
+          self.denied_list.insert(WritePermission(resolved_path));
+          self.global_state = PermissionState::Denied;
+          PermissionState::Denied
         }
+      } else {
+        state
       }
-      state
     } else {
-      let state = self.query_read(&None);
+      let state = self.query(&None);
       if state == PermissionState::Prompt {
-        if permission_prompt("Deno requests read access") {
-          self.read.granted_list.clear();
-          self.read.global_state = PermissionState::Granted;
-          return PermissionState::Granted;
+        if permission_prompt("write access") {
+          self.granted_list.clear();
+          self.global_state = PermissionState::Granted;
+          PermissionState::Granted
         } else {
-          self.read.global_state = PermissionState::Denied;
-          return PermissionState::Denied;
+          self.global_state = PermissionState::Denied;
+          PermissionState::Denied
         }
+      } else {
+        state
       }
-      state
     }
   }
 
-  pub fn request_write(&mut self, path: &Option<&Path>) -> PermissionState {
+  pub fn revoke(&mut self, path: &Option<&Path>) -> PermissionState {
     if let Some(path) = path {
-      let (resolved_path, display_path) = self.resolved_and_display_path(path);
-      let state = self.query_write(&Some(&resolved_path));
-      if state == PermissionState::Prompt {
-        if permission_prompt(&format!(
-          "Deno requests write access to \"{}\"",
-          display_path.display()
-        )) {
-          self
-            .write
-            .granted_list
-            .retain(|path| !path.starts_with(&resolved_path));
-          self.write.granted_list.insert(resolved_path);
-          return PermissionState::Granted;
-        } else {
-          self
-            .write
+      let path = resolve_from_cwd(path).unwrap();
+      self
+        .granted_list
+        .retain(|path_| !path_.0.starts_with(&path));
+    } else {
+      self.granted_list.clear();
+      if self.global_state == PermissionState::Granted {
+        self.global_state = PermissionState::Prompt;
+      }
+    }
+    self.query(path)
+  }
+
+  pub fn check(&self, path: &Path) -> Result<(), AnyError> {
+    let (resolved_path, display_path) = resolved_and_display_path(path);
+    self.query(&Some(&resolved_path)).check(
+      &format!("write to \"{}\"", display_path.display()),
+      self.name,
+    )
+  }
+}
+
+#[derive(Clone, Eq, PartialEq, Hash, Debug, Default, Deserialize)]
+pub struct NetPermission(pub String, pub Option<u16>);
```

Comment:
Nitpick: might make more sense to have named fields?

Reply:
Made this tuple just to alignments to what is supplied as arguments in the other net calls, so tuple. I can change it though if you prefer

Reply:
Sounds reasonable